v2/cli/init_command.go

```diff
@@ -0,0 +1,248 @@
+	Usage: "Initialize a template Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to create " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Set the language of the brigade project file" +
+				"to either JavaScript or TypeScript (default)",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage:   "Initialize the brigade project from an existing git repo",
+		},
+	},
+	Action: initialize,
+}
+
+// Handle init CLI flags
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := InitFields{
+		ID:       c.String(flagID),
+		Language: strings.ToLower(c.String(flagLanguage)),
+		Git:      c.String(flagGit),
+	}
+
+	err := idMatch(fields.ID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = normalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	if dirExists, dirErr := file.Exists(".brigade"); dirErr != nil {
+		return dirErr
+	} else if !dirExists {
+		fmt.Println("Creating .brigade directory...")
+		err = os.Mkdir(".brigade", 0755)
+		if err != nil {
+			return err
+		}
+	}
+
+	buf, err := execTemplate(newProjectFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "project.yaml"),
+		"project.yaml",
+		buf,
+	); err != nil {
+		return err
+	}
+
+	// If git repo is provided, create a separate js/ts file
+	if fields.Git != "" {
+		gitBuf, gitErr :=
+			execTemplate(newGitBrigadeFileBytes, fields)
+		if gitErr != nil {
+			return gitErr
+		}
+
+		gitFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+		if err = checkFileOverwrite(
+			path.Join(".brigade", gitFileName),
+			gitFileName,
+			gitBuf,
+		); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Right now the notes doesn't have template fields, add later!
+	notesBuf, err :=
+		execTemplate(newNotesFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "NOTES.txt"),
+		"NOTES.txt",
+		notesBuf,
+	); err != nil {
+		return err
+	}
+
+	secretsBuf, err :=
+		execTemplate(newSecretsFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "secrets.yaml"),
+		"secrets.yaml",
+		secretsBuf,
+	); err != nil {
+		return err
+	}
+
+	err = addToGitignore(".brigade/secrets.yaml")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Detect if .gitignore exists. If it does, add .brigade/secrets.yaml to it
+// If not, create it and add .brigade/secrets.yaml to it
+func addToGitignore(filepath string) error {
```

Comment:
`filepath` may easily be misinterpreted below as the path of the `.gitignore` file. It takes careful reading of the code to realize otherwise. A name like `pathToIgnore` might be less ambiguous.